# BSIAB_CC92-ST208_in_China
 This repository contains the analysis scripts that were written and used in the article of "**National genomic epidemiology and phylodynamics of Acinetobacter baumannii bloodstream isolates in China: the population superseding and toxification within CC92**".
 
 The links below the sub-headings lead to the scripts needed for the corresponding steps. Most of the scripts were developed for running on the Ubuntu 18.04.6 LTS server. You may download and adapt the scripts to suit your own requirements.

## 1. Software used in this workflow
- [R](https://www.r-project.org/)
- [Perl](https://www.perl.org/)
- [Python3](https://www.python.org/)
- [PRINSEQ-lite](https://github.com/uwb-linux/prinseq)
- [SPAdes](https://github.com/ablab/spades)
- [QUAST](https://github.com/ablab/quast)
- [FastANI](https://github.com/ParBLiSS/FastANI)
- [Prokka](https://github.com/tseemann/prokka)
- [mlst](https://github.com/tseemann/mlst)
- [RAxML](https://evomics.org/learning/phylogenetics/raxml/)
- [Gubbins](https://github.com/nickjcroucher/gubbins)
- [BEAST2](https://www.beast2.org/)
## 2. Reads trimming
### PRINSEQ-lite
``

## 3. Assembly
### SPAdes
``

## 4. Genome assembly evaluation
### QUAST
``

## 5. Amino acid identity (ANI) calculation
### FastANI
``


## 6. Genome annotation
### Prokka
``

## 7. ST identification and diversity analysis
### mlst
``

### diversity analysis
Diversity indices of ST s were calculated via a custom script available on GitHub (https://github.com/ahmedmagds/rarefaction-curves).

## 8. 




